### ISSUE TITLE
Add labelFormat to show path segments that are different

### ISSUE THREAD

**Issue by petrdanecek** [Microsoft/vscode#44252]
Is there any way how to have the FileName distinguished by the different part of path.
I mean I do have this default setting:
 "workbench.editor.labelFormat": "default",

which results in:
/workspace/frontend/pageName/modules/moduleName/components/VerySameFileName.jsx
/workspace/frontend/pageName/modules/moduleName/containers/VerySameFileName.jsx

Above is just example, but having few of such a TABs eats all my screen space. What I would appreciate is to have the option like the difference part of path only...like:
.../components/VerySameFileName.jsx
.../containers/VerySameFileName.jsx

.../moduleNameA/components/VerySameFileName.jsx
.../moduleNameB/components/VerySameFileName.jsx

Thanks for great VSCODE..really love it!

**Comment by bpasero** [comment]
@petrdanecek can you show a screenshot?

**Comment by petrdanecek** [comment]
![image](https://user-images.githubusercontent.com/29194829/36584278-5ce3cf10-1879-11e8-89ae-745436ca4fb2.png)

**Comment by petrdanecek** [comment]
About screenshot is for two tabs only, obviously I do have many of them

**Comment by bpasero** [comment]
@petrdanecek but you have configured the `workbench.editor.labelFormat` setting right?

**Comment by petrdanecek** [comment]
I started with not user customization. Today I added to my User Settings:
"workbench.editor.labelFormat": "default",

I tested all other options and the closest approach is the "short". Unfortunately it does work well just in case of the difference in very parent directory. Once the difference is higher in the path it is a bit mess:
screenshot of defaults:
![image](https://user-images.githubusercontent.com/29194829/36589014-3044b602-188a-11e8-8ade-1811f0ab6df1.png)

screenshot of short:
![image](https://user-images.githubusercontent.com/29194829/36589042-46d91e12-188a-11e8-91b5-0fd95eb4127b.png)

I would prefer some other option similar to short which compares and displays the different part of the path. In this case it would be:
../common/modules/Notifier/components/
../innotas/modules/Notifications/components/

so the m:/frontend is ommited as it is shared part of both files.

These are just examples, the directory nesting can be complicated..
The point is to display the different part of the path only.
